lib/hutch/worker.rb

```diff
@@ -129,11 +130,23 @@ def handle_error(message_id, payload, consumer, ex)
       end
     end
 
+    def acknowledge_error(delivery_info, properties, broker, ex)
+      acks = error_acknowledgements +
+        [Hutch::Acknowledgements::NackOnAllFailures.new]
```

Comment:
Answering my own question: because `:error_acknowldgements` is not a list, it's a chain of responsibility. Why so? Because double-acking would result in a channel exception and make underlying channel useless.